torch/_dynamo/variables/lists.py

```diff
@@ -550,7 +550,6 @@ def as_python_constant(self):
         )
 
     def reconstruct(self, codegen: "PyCodegen") -> None:
-        assert "deque" not in codegen.tx.f_globals
```

Comment:
Can you add a test where `deque` is shadowed?  E.g. `deque=None` in global scope.

I think that is what this assert was worried about and I want to make sure it isn't an issue.

Reply:
I add a test for it.